teps/0059-skip-guarded-task-only.md

```diff
@@ -249,6 +287,267 @@ Users should be able to specify that a guarded `Task` only should be skipped whe
+    expressions:
+    - input: $(params.git-action)
+      operator: in
+      values:
+      - merge
+  taskRef:
+  - name: manual-approval 
+
+- name: slack-msg
+  params:
+  - name: approver
+    value: $(tasks.manual-approval.results.approver)
+  taskRef:
+  - name: slack-msg
+  
+- name: build-image
+  runAfter:
+  - manual-approval  
+  taskRef:
+  - name: build-image
+
+- name: deploy-image
+  runAfter:
+  - build-image
+  taskRef:
+  - name: deploy-image
+  
+```
+
+If the `WhenExpressions` in `manual-approval` evaluate to `False`, then `manual-approval` would be skipped and:
+- `build-image` and `deploy-image` would be executed
+- `slack-msg` would be skipped due to missing resource from `manual-approval`
+
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+`WhenExpressions` already have tests for validation and functionality. We will add unit tests and integration tests for validation and functionality of the new syntax for full coverage.
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility 
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+### Reusability
+By unblocking the execution of dependent `Tasks` when a guarded `Task` is skipped, we enable execution to continue when the guarded `Task` is either successful or skipped, making the `Pipeline` more reusable for more scenarios or use cases
+
+### Simplicity
+
+By scoping the skipping strategy to `WhenExpressions` only, we provide the flexibility safely with a minimal change. We also limit the interleaving of `Pipeline` graph execution paths and maintain the simplicity of the workflows.
+
+### Flexibility
+
+This proposal gives users more control to specify what happens when each guarded `Task` is skipped. Moreover, changing the structure under `when` from a `list` to a `dictionary` makes it extensible because we can add more fields in the future if needed. 
+
+## Upgrade & Migration Strategy
+
+<!--
+Use this section to detail wether this feature needs an upgrade or
+migration strategy. This is especially useful when we modify a
+behavior or add a feature that may replace and deprecate a current one.
+-->
+
+To enable users to smoothly transition from the old syntax to the new syntax, we will initially support both sytanxes. Then we will deprecate the old syntax and support the new syntax only as we go towards v1 API.
```

Comment:
I think it would be good to make a more detailed plan about when we need to deprecate the old syntax.
Would the new syntax enter the API as alpha? If so what would be the plan to move to beta and deprecate the old syntax?

Reply:
will review the feature gates and add more details here